Sources/Queryable/Handlers/Queryable+Conditional.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  IfQueryable.swift
+//
+//
+//  Created by Kai Quan Tay on 12/1/24.
+//
+
+import SwiftUI
+
+public struct WithQuery<Item, Result, QueryContent: View>: View {
+    @ObservedObject private var queryable: Queryable<Item, Result>
+    private var queryContent: (_ item: Item, _ query: QueryResolver<Result>) -> QueryContent
+
+    public init(
+        _ queryable: Queryable<Item, Result>,
+        animation: Animation? = nil,
+        alignment: Alignment = .center,
+        @ViewBuilder queryContent: @escaping (_ item: Item, _ query: QueryResolver<Result>) -> QueryContent
+    ) {
+        self.queryable = queryable
+        self.animation = animation
+        self.alignment = alignment
+        self.queryContent = queryContent
+    }
```

Comment:
Sorry, this one's on me 😅 I used this "suggested change" feature but didn't include the intializer. I don't think this even compiles right now

```suggestion
    public init(
        _ queryable: Queryable<Item, Result>,
        @ViewBuilder queryContent: @escaping (_ item: Item, _ query: QueryResolver<Result>) -> QueryContent
    ) {
        self.queryable = queryable
        self.queryContent = queryContent
    }
```